### PR TITLE
Integrate cell-types-eval nested structure 

### PR DESCRIPTION
**Pull request by rubenchazarra**
This PR contains the work done over the past weeks:
**DONE:**
- [x] Include process to average the different fold's output tables.
- [x] Submodule fetching script. 
- [x] Add data download step
- [x] Improve parametrization for compatibility with nested workflows updates

**TODO:**
- [ ] Simplify **label-analysis** output renaming and collection into a single process 
- [ ] Potential better structure of publishDirs

### CODE REVIEW

**Comment by pinin4fjords** [comment]
Thanks @rubenchazarra - this is a bit fiddly, but looks like you're going the right way by passing the RESULTS_DIR. The child workflow then needs to use that value in publishDir directives.

@a-solovyev12 - could you work with Ruben to make this work please? 

For further examples, see how https://github.com/ebi-gene-expression-group/scxa-control-workflow calls child workflows, for example https://github.com/ebi-gene-expression-group/scxa-smartseq-quantification-workflow.

**Comment by a-solovyev12** [comment]
@rubenchazarra, as @pinin4fjords suggested, you can make [this](https://github.com/ebi-gene-expression-group/cell-types-eval-workflows) repo a submodule of [this one](https://github.com/ebi-gene-expression-group/cell-types-eval-control-workflow). Then you will need to clone the second one into your cross-val repository and recursively update it before running `main.nf`. Use [this script](https://github.com/ebi-gene-expression-group/cell-types-eval-control-workflow/blob/develop/bin/fetch-tool-eval-workflows.sh) as an example.

**Comment by a-solovyev12** [comment]
@rubenchazarra The figure looks awesome, thanks! As soon as @pinin4fjords approves the PR, please merge into develop

**Review by a-solovyev12 — APPROVED** [review]

File: bin/avg-tables-cell-label-analysis.R [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env Rscript
```
Comment (pinin4fjords) [inline bin/avg-tables-cell-label-analysis.R:1]: Sorry to be a nuisance guys- but should this maybe be a part of Andrey's Bioconda package? Will make it part of a container which will make life easier for us later.

**Comment by pinin4fjords** [comment]
Good work @rubenchazarra - could you just look at the refactoring mentioned above? Also don't forget to tweak the PR title if you actual do want to merge ;-)

**Comment by a-solovyev12** [comment]
@pinin4fjords that sounds like a good idea. I can add it to the next release of cell-types-analysis, when more changes accumulate

**Comment by pinin4fjords** [comment]
@rubenchazarra - could you do some tidying up? Remove anything you've commented, fix indentations etc? Will make review easier.

**Comment by a-solovyev12** [comment]
Looks good to me @rubenchazarra. After implementing what @pinin4fjords suggested, feel free to merge. Thanks!

**Review by a-solovyev12 — APPROVED** [review]

**Pull request merged**